### ISSUE TITLE
Potential bug in `random_int` from `prng_ffi.mjs`

### ISSUE THREAD

**Issue by mooreryan** [giacomocavalieri/prng#11]
Hi! While investigating this: https://github.com/mooreryan/gleam_qcheck/issues/7, I think I have run across a bug in the `random_int` function from `prng_ffi.mjs`.

I suspect that this line: https://github.com/giacomocavalieri/prng/blob/c9275a232efe10b21cf7ad74187b49f55ff59ac8/src/prng_ffi.mjs#L34, 

should rather be `iteration_seed = next(iteration_seed)`.

Here is the function:

```js
export function random_int(seed, from, to) {
    const range = to - from + 1
    const is_power_of_2 = ((range - 1) & range) === 0
    if (is_power_of_2) {
        const number = ((range - 1) & seed_to_int(seed)) >>> 0
        return [number + from, next(seed)]
    } else {
        const threshold = (((-range) >>> 0) % range) >>> 0
        let iteration_seed = seed
        let x = undefined
        do {
            x = seed_to_int(iteration_seed)
            iteration_seed = next(seed)
        } while (x < threshold)
        return [from + (x % range), iteration_seed]
    }
}
```

Here is a thought experiment.  Take a case where:

- the `do` loop goes 1 time, and produces an `x` such that `x < threshold`.  A new `iteration_seed` is generated from `seed`, let's call it `5`.
- Now the `do` loop goes for a second time.  Suppose that it again produces an `x` such that `x < threshold`.  A "new" iteration seed is generated from `seed`.  However since `seed` wasn't updated, the same `iteration_seed` will be generated (in this example, it is `5`).
- Now the `do` loop goes for a third time.  Because we have the same `iteration_seed` of `5`, the same `x` will be generated, and `x < threshold` will be true. 
- And, we have created an infinite loop.

I have observed that changing `iteration_seed = next(seed)` to  `iteration_seed = next(iteration_seed)` addresses this issue (and fixes the original issue in qcheck).  Do you think this is the appropriate change?

**Comment by giacomocavalieri** [comment]
Hello! Thank you so much for the investigative work, it's super valuable!! I'll have to check the original algorithm and see if I've messed up the Gleam translation to be sure but it looks like it's most likely an error on my part

**Comment by giacomocavalieri** [comment]
You are totally right, that's a bug in the JS implementation. I've released v4.0.1 fixing it; thanks again!
